plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/model/order/LineItem.kt

```diff
@@ -38,4 +38,17 @@ data class LineItem(
             Attribute(it.displayKey, it.displayValue as String)
         } ?: emptyList()
     }
+
+    fun getConfigurationKey(): Long? {
+        return when {
+            bundledBy.isNullOrBlank().not() -> {
+                (metaData?.first { meta ->
+                    meta.key == WCMetaData.BundleMetadataKeys.BUNDLED_ITEM_ID
+                }?.value.toString())
+                    .toLongOrNull()
+            }
+
+            else -> null
+        }
```

Comment:
NP: Using the `takeIf` can simplify your code since it already implies that if the condition is not met, a `null` will be returned, also, this looks like a computed property scenario, as the function is described as a getter:

```suggestion
    val configurationKey
        get() = bundledBy.takeIf { it.isNullOrBlank().not() }
            ?.let { metaData }
            ?.find { it.key == WCMetaData.BundleMetadataKeys.BUNDLED_ITEM_ID }
            ?.value.toString().toLongOrNull()
```